libs/prov/prov_k8s_cortx_deploy.py

```diff
@@ -1597,6 +1554,31 @@ def service_upgrade_software(node_obj: LogicalNode, upgrade_image_version: str)
             PROV_CFG['k8s_cortx_deploy']["k8s_dir"]), read_lines=True)
         return resp
 
+    def pre_check(self, master_node_list):
+        """
+        This method will dump all the info before deployment starts
+        It will capture the taint nodes is any stale entries left out like
+        services or any other resources.
+        Param: master_node_list: node obj for master node.
+        returns true, resp_list
+        """
+        taint_cmd = common_cmd.KUBECTL_GET_TAINT_NODES.format(self.deploy_cfg["pre_check_log"])
+        all_resource = common_cmd.KUBECTL_GET_ALL.format(self.deploy_cfg["pre_check_log"])
+        get_secret = common_cmd.KUBECTL_GET_SECRET.format(self.deploy_cfg["pre_check_log"])
+        get_pv = common_cmd.KUBECTL_GET_PV.format(self.deploy_cfg["pre_check_log"])
+        get_pvc = common_cmd.KUBECTL_GET_PVC.format(self.deploy_cfg["pre_check_log"])
+        list_pre_check = [taint_cmd, all_resource, get_secret, get_pvc, get_pv]
+        LOGGER.info("======== Running Pre-checks before deployment ==========")
+        for cmd in list_pre_check:
+            resp = master_node_list.execute_cmd(cmd, read_lines=True)
+        local_path = os.path.join(LOG_DIR, LATEST_LOG_FOLDER, self.deploy_cfg["pre_check_log"])
+        if master_node_list.path_exists(self.deploy_cfg["pre_check_log"]):
+            master_node_list.copy_file_to_local(remote_path=self.deploy_cfg["pre_check_log"],
+                                                local_path=local_path)
+            master_node_list.execute_cmd(
+                common_cmd.CMD_REMOVE_DIR.format(self.deploy_cfg["pre_check_log"]))
+        return True, resp
```

Comment:
which resp u r returning here? it will resp for last cmd run , isn't it?
do we want to return pre_chk_log instead or rather its path?

Reply:
will return the path.